src/common/services/melon/resolve.ts

```diff
@@ -13,15 +12,15 @@ const getTitle = (document_: Document) => {
 const getArtists = (document_: Document) => {
   return [...document_.querySelectorAll('.section_info .info .artist a')]
     .map((element) => element.textContent?.trim())
-    .filter(isDefined)
+    .filter((a) => a != undefined)
```

Comment:
this is a case where we need `isDefined`

TS currently isn't smart enough to realize that `a != undefined` is filtering out all undefined elements, so it thinks the return type of this function is `(string | undefined)[]`

if we use the `isDefined` type guard, TS can correctly infer the return type. it also removes the need for the cast to `ResolveData` on line 86